mass_mailing_import_mailing_list/wizard/import_mailing_list.py

```diff
@@ -0,0 +1,360 @@
+        mailing_list_ids = []
+        for mail in active_mails:
+            ml = mailing_list.filtered(lambda l: l.adkd_mail_name == mail)
+            if not ml:
+                ml = mailing_list.create({
+                    'name': f'{mail} placeholder name',
+                    'is_public': True,
+                    'adkd_mail_name': mail,
+                })
+            ml.parent_id = self.get_campaign() \
+                if self.import_type == 'adkd' else ''
+            mailing_list_ids.append(ml.id)
+        return mailing_list_ids
+
+    def insert_mail_contacts_to_mailing_lists(self,
+                                              mailing_lists_ids,
+                                              contacts):
+        """
+        Clears all contacts form the
+        mailing lists and adds the imported contacts
+        unless they have opted out for adkd campaign
+        """
+        mailing_lists = self.env['mail.mass_mailing.list'].browse(
+            mailing_lists_ids
+        )
+        for mail_list in mailing_lists:
+            mail_list.write({
+                'contact_ids': [(5, )]
+            })
+        list_contact = self.env['mail.mass_mailing.list_contact_rel']
+        for mail_name, contact_id in contacts:
+            mail_list = mailing_lists.filtered(
+                lambda l: l.adkd_mail_name.upper() == mail_name)
+            if mail_list:
+                list_contact_id = list_contact.search(
+                        [('contact_id', '=', contact_id),
+                         ('list_id', '=', mail_list.parent_id.id
+                         if self.import_type == 'adkd'
+                         else mail_list.id)]
+                    )
+                if list_contact_id and list_contact_id.opt_out:
+                    continue
+                mail_list.write({
+                    'contact_ids': [(4, contact_id)],
+                })
+                if self.import_type == 'adkd':
+                    mail_list.parent_id.write({
+                        'contact_ids': [(4, contact_id)],
+                    })
+
+    def import_data(self):
+        """Parse and import a file."""
+        if not self.file or not \
+                self.filename.lower().endswith(
+                    ('.xls', '.xlsx', '.csv', 'txt')):
+            raise UserError(
+                _("Please Select an .xls, .xlsx, "
+                  ".txt or .csv file to Import.")
+            )
+        partner_obj = self.env['res.partner']
+        contact_obj = self.env['mail.mass_mailing.contact']
+        failed_imports = []
+        contacts = []
+        active_mails = set()
+        if self.filename.lower().endswith(('.csv', '.txt')):
+            # Decode data to string, Odoo supplies it as base64
+            csv_data = base64.b64decode(self.file)
+            try:
+                data_file = io.StringIO(csv_data.decode("ISO-8859-1"))
+            except Exception as e:
+                raise UserError('Error %s' % e)
```

Comment:
Please use f'strings' or 'string'.format() as this string formatting syntax is depricated